### PR TITLE
Templates

### PR DESCRIPTION
**Pull request by JW-Vinayak**
Hi Andreas,

Major changes are in spec folder only. Other changes are related to code formatting.

### CODE REVIEW

File: spec/common.js [hunk 1]
```diff
@@ -36,7 +36,7 @@
     var text = "";
     var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
 
-    for( var i=0; i < 15; i++ )
+    for (var i = 0; i < 15; i++)
       text += possible.charAt(Math.floor(Math.random() * possible.length));
```
Comment (argelius) [inline spec/common.js]: Please use curly brackets even if the body is just one line.

File: src/monaca.js [hunk 2]
```diff
@@ -1189,143 +1190,191 @@
...
+     *   and will install all the npm packages.
+     * @param {object} template - template parameters 
+     * @param {string} template.name - name of the template
+     * @param {string} template.path - path from where the template will be taken
+     * @param {string} appname - name with which the app will be created and an entry will be made into parmanent storage
+     * @param {string} workingDir - where on disk the app will be created.
+     * @return {Promise}
+     * @example
+     *    var template = {
+     *       name: 'Onset Tab Bar',
```
Comment (argelius) [inline src/monaca.js]: Spelling

File: src/monaca.js [hunk 3]
```diff
@@ -1189,143 +1190,191 @@
...
-        npmProcess.stdout.on('data', function(data) {
-          console.log(data);
+      deferred.resolve(list);
+    } catch (e) {
+      deferred.reject(e);
+    }
+    return deferred.promise;
+  };
+
+  /**
```
Comment (argelius) [inline src/monaca.js]: Formatting

File: src/monaca.js [hunk 4]
```diff
@@ -1189,143 +1190,191 @@
...
-
-  return deferred.promise;    
-};
+    return deferred.promise;
+  };
+
+  Monaca.prototype._replaceTemplate = function(dirName, template, workingDir) {
+    var deferred = Q.defer();
+
+    async.series([
```
Comment (argelius) [inline src/monaca.js]: The indentation here looks a bit off. Can you make it more clear?

**Comment by JW-Vinayak** [comment]
Hi Andreas, I have pushed a new commit correcting all the points that you had mentioned.

**Comment by argelius** [comment]
Thanks @JW-Vinayak !

I've merged it now.

**Pull request merged**